test/datetime.test.js

```diff
@@ -0,0 +1,93 @@
+// Copyright IBM Corp. 2012,2016. All Rights Reserved.
+// Node module: loopback-connector-mysql
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+'use strict';
+
+var DateString = require('../node_modules/loopback-datasource-juggler/lib/date-string');
+var fmt = require('util').format;
+var should = require('./init.js');
+
+var db, Order, Person;
+describe('MySQL datetime handling', function() {
+  var personDefinition = {
+    name: String,
+    gender: String,
+    married: Boolean,
+    dob: DateString,
+    createdAt: {type: Date, default: Date},
+  };
+
+  // Modifying the connection timezones mid-flight is a pain,
+  // but closing the existing connection requires more effort.
+  function setConnectionTimezones(tz) {
+    db.connector.client._allConnections.forEach(function(con) {
+      con.config.timezone = tz;
+    });
+  }
+  before(function(done) {
+    db = getSchema({
+      dateStrings: true,
+    });
+    Person = db.define('Person', personDefinition, {forceId: true, strict: true});
+    db.automigrate(['Person'], done);
+  });
+
+  beforeEach(function() {
+    setConnectionTimezones('Z');
+  });
+  after(function(done) {
+    Person.destroyAll(function(err) {
+      db.disconnect(function() {
+        return done(err);
+      });
+    });
+  });
+
+  it('should allow use of DateStrings', function(done) {
+    var d = new DateString('1971-06-22');
+    var p = Person.create({
+      name: 'Mr. Pink',
+      gender: 'M',
+      dob: d,
+      createdAt: new Date(),
+    }).then(function(inst) {
+      inst.should.not.eql(null);
+      inst.dob.toString().should.eql(d.toString());
```

Comment:
I might be wrong, not sure does `create` return the same data in cb as `find` returns. According to the next test, you want to test the data returned by `find` :)